vagrant/Vagrantfile

```diff
@@ -0,0 +1,81 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+
+# Provisioning script to install Docker and dcos-e2e
+$provision_script = <<SCRIPT
+#!/bin/bash
+
+set -o errexit -o nounset -o pipefail -x
+
+# Docker CE installation instructions
+apt-get update
+apt-get install -y apt-transport-https ca-certificates curl software-properties-common
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+apt-key fingerprint 0EBFCD88 | grep -c Docker
+add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+apt-get update
+apt-get install -y docker-ce=17.12.0~ce-0~ubuntu
+
+# Allow user to run Docker commands
+usermod -a -G docker ubuntu
+
+# Install dcos-e2e
+apt-get install -y python3-pip
+pip3 install git+https://github.com/mesosphere/dcos-e2e.git@master
+
+# Accept IP forwarding to Docker containers.  On the VM host, set this VM guest's IP (192.168.18.2)
+# as the gateway for routing 172.17.0.0/16.  This allows the VM host to connect to Docker containers,
+# to use the DC/OS web interface, for example.
+iptables -I DOCKER-USER ! -i docker0 -o docker0 -j ACCEPT
+SCRIPT
+
+# Provisioning script to start user in /vagrant directory
+$user_script = <<SCRIPT
+#!/bin/bash
+
+set -o errexit -o nounset -o pipefail -x
+
+grep -q 'cd /vagrant' ~/.bash_profile || echo -e '\n[ -d /vagrant ] && cd /vagrant' >> ~/.bash_profile
+SCRIPT
+
+unless Vagrant.has_plugin?("vagrant-persistent-storage")
+  raise 'Please run "vagrant plugin install vagrant-persistent-storage" first'
+end
+
+Vagrant.configure("2") do |config|
+
+  if Vagrant.has_plugin?("vagrant-cachier")
+    config.cache.scope = :box
+  end
+  if Vagrant.has_plugin?('vagrant-vbguest')
+    config.vbguest.auto_update = true
+  end
+
+  config.vm.box = "ubuntu/xenial64"

```

Comment:
What do you think of using a box with Docker already installed? https://github.com/William-Yeh/docker-enabled-vagrant ?

Reply:
The thing I like about installing the software in the Vagrantfile is that when you do find yourself in a strange and unexpected variation of OS/network setup, it's quick to modify the configuration to get around problems.

I'd suggest a good follow-up could be to add PowerShell and bash scripts to build the box once, and save it locally. This gives the speed of a pre-built box with the flexibility of having the `Vagrantfile` right there.

Reply:
😄 I'll leave it to you to decide what's best. Feel free to make any JIRA issues you like for this.